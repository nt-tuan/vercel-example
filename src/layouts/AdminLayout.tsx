import {
  AppBar,
  Box,
  Breadcrumbs,
  BreadcrumbsProps,
  Toolbar,
  Typography,
} from "@mui/material";
import React from "react";
interface Props {
  children: React.ReactNode;
}

export const AdminLayout = (props: Props) => {
  return (
    <Box sx={{ backgroundColor: "#fafafa", minHeight: "100%" }}>
      <AppBar position="static">
        <Toolbar>
          {/* <IconButton
            size="large"
            edge="start"
            color="inherit"
            aria-label="menu"
            sx={{ mr: 2 }}
          >
            <MenuIcon />
          </IconButton> */}
          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            Distichain
          </Typography>
        </Toolbar>
      </AppBar>
      <Box sx={{ marginTop: "24px", marginLeft: "20px", marginRight: "20px" }}>
        {props.children}
      </Box>
    </Box>
  );
};

AdminLayout.Header = ({ children }: { children: React.ReactNode }) => {
  return (
    <Typography variant="h4" component="h2">
      {children}
    </Typography>
  );
};

AdminLayout.Breadcrumbs = (props: BreadcrumbsProps) => {
  return (
    <Box sx={{ marginTop: "8px", marginBottom: "24px" }}>
      <Breadcrumbs {...props} />
    </Box>
  );
};

AdminLayout.Content = ({ children }: { children: React.ReactNode }) => {
  return <Box sx={{ marginTop: "24px" }}>{children}</Box>;
};
