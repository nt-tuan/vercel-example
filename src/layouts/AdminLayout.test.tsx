import { render, fireEvent } from "@testing-library/react";
import { AdminLayout } from "./AdminLayout";

jest.mock("./SideMenu", () => ({
  SideMenu: () => <div>side-menu</div>,
}));

test("AdminLayout should work", () => {
  const { getByText, getByTestId } = render(
    <AdminLayout>children</AdminLayout>
  );
  fireEvent.click(getByTestId("ChevronLeftIcon"));
  expect(getByText("children")).toBeInTheDocument();
});
