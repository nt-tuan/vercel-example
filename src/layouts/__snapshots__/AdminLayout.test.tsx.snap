// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLayout should work 1`] = `
<div>
  <div
    class="MuiBox-root css-18qj8iw"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          Distichain
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root css-7gojc3"
    >
      <h2
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        My Header
      </h2>
      <div
        class="MuiBox-root css-r4f27e"
      >
        <nav
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="/link1"
              >
                Link 1
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="MuiBox-root css-178yklu"
      >
        Content
      </div>
    </div>
  </div>
</div>
`;
